### PR TITLE
Fix custom ad notification z-ordering and possible display issues.

### PR DESCRIPTION
**Pull request by aseren**
Resolves https://github.com/brave/brave-browser/issues/22203

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

The issue should be tested on Windows and MacOS platforms.

- Clean install
- Run Brave with: `--enable-features="CustomAdNotifications:should_attached_ad_notification_to_browser_window/true"`
- Enable rewards and ads
- Trigger a notification ad
- Check that the notification ad has the same Z-order as the Browser window

### CODE REVIEW

File: browser/ui/brave_ads/notification_ad_popup_handler.h [hunk 1]
```diff
@@ -46,11 +46,6 @@ class NotificationAdPopupHandler final {
   // |by_user| is true if the notification popup was closed by the user,
   // otherwise false.
   static void Close(const std::string& notification_id, bool by_user);
-
-  // Move the notification popup view for the given |notification_id| by a
-  // distance.
```
Comment (tmancey) [inline browser/ui/brave_ads/notification_ad_popup_handler.h:51]: nit: `|distance|`

Comment (tmancey) [reply browser/ui/brave_ads/notification_ad_popup_handler.h:51]: Oh my mistake, removing comment

File: components/brave_ads/common/features.cc [hunk 2]
```diff
@@ -62,6 +56,12 @@ constexpr char kFieldTrialParameterNotificationAdDarkModeBackgroundColor[] =
 // Default color value is SkColorSetRGB(0x20, 0x23, 0x27);
 constexpr char kDefaultNotificationAdDarkModeBackgroundColor[] = "202327";
 
+// Set to true to support multiple displays or false to only support the primary
+// display
```
Comment (tmancey) [inline components/brave_ads/common/features.cc:60]: nit: `.` at end of sentence

**Review by tmancey — APPROVED** [review]
LGTM++

File: components/brave_ads/common/features.cc (hunk 2 above)
Comment (aseren) [reply components/brave_ads/common/features.cc:60]: All other comments in this file don't have `.` at the end of a sentence. Maybe it is better to fix all of them as a followup?

Comment (tmancey) [reply components/brave_ads/common/features.cc:60]: Sounds good to me

Comment (tmancey) [reply components/brave_ads/common/features.cc:60]: I will raise a PR over the weekend as in the process of adding some public facing comments anyhow. Thanks

**Pull request merged**